File: Python/program/Car.py
#Saya Nur Ainun 2202046 mengerjakan LATIHAN PRAKTIKUM 4 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
#untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin. 

from Vehicle import Vehicle

class Car(Vehicle):
    __jumlahKursi = ""
    __jumlahPintu = ""

    #constructor
    def __init__(self,  platNomor, merk, tahunProduksi, warna, jumlahKursi, jumlahPintu):
        super().__init__(platNomor, merk, tahunProduksi, warna)
        self.__jumlahKursi = jumlahKursi
        self.__jumlahPintu = jumlahPintu

    # Getter and Setter jumlahKursi
    def getJumlahKursi(self):
        return self.__jumlahKursi
    
    def setJumlahKursi(self, jumlahKursi):
        self.__jumlahKursi = jumlahKursi 

    # Getter and Setter jumlahPintu
    def getJumlahPintu(self):
        return self.__jumlahPintu 
    
    def setJumlahPintu(self, jumlahPintu):
        self.__jumlahPintu = jumlahPintu

File: C++/program/Car.cpp
/* Saya Nur Ainun 2202046 mengerjakan Latihan Praktikum 4 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin. */


#pragma once
// memasukan liblary yang di gunakan termasuk kelas induk (kelas Vehicle)
//dari kelas car

#include <iostream>
#include <string>
#include "Vehicle.cpp"


// deklarasi namespace
using namespace std;

class Car : public Vehicle
{
private:
    int jumlah_kursi;
    int jumlah_pintu;

public:

    Car(string plat_nomor = "", string merk = "", string tahun_produksi = "", string warna = "",int jumlah_kursi = 0 , int jumlah_pintu = 0) 
    : Vehicle (plat_nomor, merk, tahun_produksi, warna)
    {
        this->jumlah_kursi = jumlah_kursi;
        this->jumlah_pintu = jumlah_pintu;
    } 

    void setJumlah_kursi(int jumlah_kursi)
    {
        this->jumlah_kursi = jumlah_kursi;
    }

    void setJumlahPintu(int jumlah_pintu)
    {
        this->jumlah_pintu = jumlah_pintu;
    }

    int getjumlah_Kursi()
    {
        return jumlah_kursi;
    }

    int getJumlahPintu()
    {
        return jumlah_pintu;
    }

    void displayInfo_Car() {
        Vehicle::displayInfo();
        cout << "\nJumlah Kursi: " << jumlah_kursi << "\nJumlah Pintu: " << jumlah_pintu;
    }

    //Destructor
    ~Car()
    {

    }

};